### ISSUE TITLE
QC by mitochondrial content fails if no filtering is chosen during initial step

### ISSUE THREAD

**Issue by JPReceveur** [IGS/gEAR#35]
Using the single cell workbench, if no filtering boxes are chosen (default shows exclude cells with <300 genes and genes in < 3 cells but boxes are not checked by default), can proceed to the next step but QC step will give an error.
Error replicated with: 
    P2, mouse, scRNA-seq, cochlea (Hertzano/Ament) 
    Crista Ampullaris, Macrophages, PseudoExp (Bermingham-McDonogh)

**Comment by adkinsrs** [comment]
This is on umgear.org, correct?  I'll peruse the logs.  May be a missing module.

**Comment by adkinsrs** [comment]
Replicated the issue (did not check any filter boxes, and used P2, mouse, scRNA-seq, cochlea (Hertzano/Ament)
dataset). Here is the error from logs

```
Traceback (most recent call last):
  File "/var/www/cgi/h5ad_qc_by_mito.cgi", line 100, in <module>
    main()
  File "/var/www/cgi/h5ad_qc_by_mito.cgi", line 75, in main
    jitter=0.4, multi_panel=True, save="_qc_by_mito.png")
  File "/opt/Python-3.7.3/lib/python3.7/site-packages/scanpy/plotting/_anndata.py", line 700, in violin
    obs_df = get.obs_df(adata, keys=keys, layer=layer, use_raw=use_raw)
  File "/opt/Python-3.7.3/lib/python3.7/site-packages/scanpy/get.py", line 165, in obs_df
    f"Could not find keys '{not_found}' in columns of `adata.obs` or in"
KeyError: "Could not find keys '['n_genes']' in columns of `adata.obs` or in `adata.var_names`."
```

It looks like the CGI script for the filtering step h5ad_apply_primary_filter.cgi adds the "n_genes" and "n_cells" in the set of observations for the AnnData object by way of Scanpy.  If no filters are applied, then these columns are not added and since they are required in the QC-by-mito step the tool fails.

What I can do is write a check where if no filters are applied, then we filter all cells that are < total_genes in the initial shape, which should get those statistics but not do any actual filtering.

**Comment by jorvis** [comment]
I'm a little surprised this could happen at all. I thought the toggle for QC by mito wasn't even enabled until a primary filter step was performed.

**Comment by adkinsrs** [comment]
@jorvis would it be a better fix to ensure that toggle is disable until at least one filter is checked?  While rare, I feel there may be times where a user would just want to proceed with the unfiltered data.

**Comment by adkinsrs** [comment]
Addressed in commit 7fa6d38

Works on my machine and will deploy and test on the devel server soon

**Comment by adkinsrs** [comment]
Deployed on the devel server.  Will close this ticket for now, as @jorvis plans on deploying the codebase to the main gEAR site within the next week or so.
